test/helpers/position_assertions.cc

```diff
@@ -1232,6 +1233,80 @@ string HoverAssertion::toString() const {
     return fmt::format("hover: {}", message);
 }
 
+shared_ptr<HoverMultilineAssertion> HoverMultilineAssertion::make(string_view filename, unique_ptr<Range> &range,
+                                                                  int assertionLine,string_view assertionContents,
+                                                                  string_view assertionType) {
+    static const regex multilineRegex(R"(^([0-9]+) (.+)$)");
+
+    smatch matches;
+    string assertionContentsString = string(assertionContents);
+
+    if (regex_search(assertionContentsString, matches, multilineRegex)) {
+        auto lineno = stoi(matches[1].str());
+        auto contents = matches[2].str();
+        return make_shared<HoverMultilineAssertion>(filename, range, assertionLine, lineno, contents);
+    }
+
+    ADD_FAIL_CHECK_AT(
+        string(filename).c_str(), assertionLine + 1,
+        fmt::format(
+            "Improperly formatted hover-multiline assertion. Expected '<lineno> <contents>'. Found '{}'",
+            assertionContents, filename));
+
+    return nullptr;
+}
+HoverMultilineAssertion::HoverMultilineAssertion(string_view filename, unique_ptr<Range> &range, int assertionLine,
+                                                 int lineno, string_view message)
+    : RangeAssertion(filename, range, assertionLine), lineno(lineno), message(string(message)) {}
+
+void HoverMultilineAssertion::checkAll(const vector<shared_ptr<RangeAssertion>> &assertions,
+                              const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                              LSPWrapper &wrapper, int &nextId, string errorPrefix) {
+    for (auto assertion : assertions) {
+        if (auto assertionOfType = dynamic_pointer_cast<HoverMultilineAssertion>(assertion)) {
+            assertionOfType->check(sourceFileContents, wrapper, nextId, errorPrefix);
+        }
+    }
+}
+
+// Returns `true` if `line` matches a full line of text in `text`.
+bool containsLineAt(string_view text, string_view line, int lineno) {
+    vector<string> lines = absl::StrSplit(text, "\n");
```

Comment:
Should this helper: 
- Exist? Does it make more sense to have this inline for the check?
- Take in the lines pre-split? That way we can split once and not have to repeat the same split across the string.

Reply:
After adding the require checks for `lineno`, it made more sense to fold this into the `check` method.